### PR TITLE
[DRAFT] Refactor map state and behavior into view model

### PR DESCRIPTION
**Pull request by gino-m**
Towards #1907

### CODE REVIEW

File: ground/src/main/java/com/google/android/ground/ui/map/CameraPosition.kt [hunk 1]
```diff
@@ -62,3 +63,7 @@ data class CameraPosition(
     }
   }
 }
+
+// TODO(#1712): Remove [isAllowZoomOut] and [bounds] from CameraPosition class.
+  fun com.google.android.gms.maps.model.CameraPosition.toModelObject(isAllowZoomOut: Boolean, bounds: Bounds) =
```
Comment (JSunde) [inline ground/src/main/java/com/google/android/ground/ui/map/CameraPosition.kt:68]: I know it's not a new addition, but should isAllowZoomOut be named allowZoomOut or isZoomOutAllowed? The current name feels a little clunky 😅

File: ground/src/main/java/com/google/android/ground/ui/map/MapViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+package com.google.android.ground.ui.map
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import kotlinx.coroutines.channels.BufferOverflow
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.asSharedFlow
+
+class MapViewModel : ViewModel() {
```
Comment (JSunde) [inline ground/src/main/java/com/google/android/ground/ui/map/MapViewModel.kt:26]: Should we add some kdoc?

File: ground/src/main/java/com/google/android/ground/ui/map/gms/GoogleMapsFragment.kt [hunk 3]
```diff
@@ -195,12 +199,13 @@ class GoogleMapsFragment : Hilt_GoogleMapsFragment(), MapFragment {
       )
     clusterManager.setOnClusterClickListener(this::onClusterItemClick)
     clusterManager.renderer = clusterRenderer
-
     map.setOnCameraIdleListener(this::onCameraIdle)
     map.setOnCameraMoveStartedListener(this::onCameraMoveStarted)
 
     map.setOnMapClickListener { onMapClick(it) }
 
+    // TODO: Initialize VM with desired MapType before calling this.
```
Comment (JSunde) [inline ground/src/main/java/com/google/android/ground/ui/map/gms/GoogleMapsFragment.kt:207]: Should we have a tracking issue for this TODO?

**Review by JSunde — APPROVED** [review]

File: ground/src/main/java/com/google/android/ground/ui/map/CameraPosition.kt (hunk 1 above)
Comment (gino-m) [reply ground/src/main/java/com/google/android/ground/ui/map/CameraPosition.kt:68]: Hey @JSunde, this PR isn't ready for review yet, it's just a WIP draft. isAllowZoomOut should be removed completely.

**Comment by gino-m** [comment]
Closing this draft for now, we can dust it off after field work is started.

**Pull request closed without merging**